core/math/quaternion.cpp

```diff
@@ -292,6 +292,19 @@ real_t Quaternion::get_angle() const {
 	return 2 * Math::acos(w);
 }
 
+Quaternion Quaternion::get_twist(const Vector3 &v) const
+{
+#ifdef MATH_CHECKS
+	ERR_FAIL_COND_V_MSG(!is_normalized(), Quaternion(), "The start quaternion must be normalized.");
+#endif
+	Vector3 p =  Vector3(x,y,z).dot(v) * v;
+	return Quaternion(p.x, p.y, p.z,w).normalized();
+}
+Quaternion Quaternion::get_swing(const Vector3 &v) const
```

Comment:
```suggestion

Quaternion Quaternion::get_swing(const Vector3 &p_axis) const
```
Space between methods